File: BackEnd-Siukat/routes/prodi.go
package routes

import (
	"BackEnd-Siukat/config"
	"BackEnd-Siukat/models"
	"net/http"

	"github.com/gin-gonic/gin"
)

func ProdiRoutes(r *gin.RouterGroup) {
	group := r.Group("/prodi")
	group.GET("/all", func(c *gin.Context) {
		var data []models.Prodi
		if err := config.DB.Find(&data).Error; err != nil {
			c.JSON(http.StatusInternalServerError, err)
			return
		}
		c.JSON(http.StatusOK, data)
	})

	group.GET("/id/:id", func(c *gin.Context) {
		id := c.Param("id")
		var data models.Prodi
		if err := config.DB.Where("kode = ?", id).First(&data).Error; err != nil {
			c.JSON(http.StatusInternalServerError, err)
			return
		}
		c.JSON(http.StatusOK, data)
	})

	group.GET("/fakultas/:kode_fak", func(c *gin.Context) {
		kodeFak := c.Param("kode_fak")
		var data []models.Prodi
		if err := config.DB.Where("kode_fak = ?", kodeFak).Find(&data).Error; err != nil {
			c.JSON(http.StatusInternalServerError, err)
			return
		}
		c.JSON(http.StatusOK, data)
	})
}

File: ZhammDataBeLama/siukat-service-dev_snmptn_2022/routes/prodi.js
var express = require('express');
var router = express.Router();
var Prodi = require('../models/prodi');
var db = require('../config/database');

router.get('/all', function(req, res, next){
  return db.transaction(function (t) {
    Prodi.findAll().then(prodi => {
      res.send(prodi)
    })
  }).catch(function (err) {
    res.status(500).json(err)
  })
})

router.get('/id/:id', function(req, res, next){
  return db.transaction(function (t) {
    Prodi.findById(req.params.id).then(prodi => {
      res.send(prodi)
    })
  }).catch(function (err) {
    res.status(500).json(err)
  })
})

router.get('/fakultas/:kode_fak', function(req, res, next){
  return db.transaction(function (t) {
    Prodi.findAll({
      where:{
        kode_fak: req.params.kode_fak
      }
    }).then(prodi => {
      res.send(prodi)
    })
  }).catch(function (err) {
    res.status(500).json(err)
  })
})

module.exports = router;
